bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -318,6 +330,77 @@ private void addWidgetMappings(EList<Mapping> mappings, UIComponent component) {
         }
     }
 
+    private void addWidgetVisibility(EList<VisibilityRule> visibility, UIComponent component) {
+        if (component.getConfig() != null && component.getConfig().containsKey("visibility")) {
+            for (Object sourceVisibility : (Collection<?>) component.getConfig().get("visibility")) {
+                if (sourceVisibility instanceof String) {
+                    String item = sourceVisibility.toString().split(COMPARATOR_STRING)[0].trim();
+                    Matcher matcher = COMPARATOR.matcher(sourceVisibility.toString());
+                    String condition = "";
+                    if (matcher.find()) {
+                        condition = matcher.group();
+                    }
+                    String sign = "";
+                    String state = sourceVisibility.toString().split(COMPARATOR_STRING)[1].trim();
+                    if ((state.charAt(0) == '+') || (state.charAt(0) == '-')) {
+                        sign = state.substring(0, 0);
+                        state = state.substring(1).trim();
+                    }
+                    VisibilityRuleImpl visibilityRule = (VisibilityRuleImpl) SitemapFactory.eINSTANCE
```

Comment:
I think this can be simplified by using a better regex pattern:

```
(?<item>[\w_]+)?\s*(?<condition>==|!=|<=|>=|<|>)\s*(?<sign>\+|-)?\s*(?<value>\S+)
```

You can then use `.find` and have direct access to all groups. I also added named groups (`item`, `condition`, `sign` and `state`), so you can use `.group("item")` instead of `.group(0)`. I believe this results in much better performance (and simpler code) than matching, splitting and trimming several times.

Did you check if this also works for the "no item name provided, use current item name" case?

A similar regex can be used for the color below.